### PR TITLE
Notify user to restart server

### PR DESCRIPTION
**Pull request by barbaravaldez**
Notify user to restart server when data and log location is changed. I had to add a delay so we don't spam the user, in case that they type the new value instead of using the browse button.

https://github.com/microsoft/azuredatastudio/assets/34872381/e044fed4-9d45-4018-a551-1c75ac2b58c2

Fix https://github.com/microsoft/azuredatastudio/issues/24306

### CODE REVIEW

**Comment by corivera** [comment]
Showing a blocking notification every time we do a validation pass seems like it would get annoying really quickly. Would you be able to show this info popup after they click Apply instead? We only need to display the reminder if actual changes have been made to the server.

**Comment by barbaravaldez** [comment]
> Showing a blocking notification every time we do a validation pass seems like it would get annoying really quickly. Would you be able to show this info popup after they click Apply instead? We only need to display the reminder if actual changes have been made to the server.

The only concern with that is that the information pop up can be easily missed, but I agree that this approach can become annoying. I think showing the information message and leaving the dialog opened would be the best option. I'll show this notification as a vscode message. Once this PR (https://github.com/microsoft/azuredatastudio/pull/23743) is completed, we can change the behavior

**Comment by corivera** [comment]
I still think using the modal option is a good idea, it just shouldn't pop up every few seconds if you're slowly working through the dialog inputs. Same with this new behavior with the notifications. There's an event you can hook into when the dialog is closed, which would be a good place to display the modal popup from for the server dialog. Here's an example of where that gets used: https://github.com/microsoft/azuredatastudio/blob/main/extensions/mssql/src/ui/dialogBase.ts#L57

So, you could register another callback for that event in the server properties dialog for displaying the restart notification.

File: extensions/mssql/tsconfig.json [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
 	"extends": "../tsconfig.base.json",
 	"compilerOptions": {
+		"experimentalDecorators": true,
```
Comment (corivera) [inline extensions/mssql/tsconfig.json:4]: Is this decorator part still needed?

**Review by corivera — APPROVED** [review]
Approved pending that last comment for the tsconfig

File: extensions/mssql/tsconfig.json (hunk 1 above)
Comment (barbaravaldez) [reply extensions/mssql/tsconfig.json:4]: oh I forgot to remove it

**Pull request merged**